example_plugins/baz/src/registerEndpoints.ts

```diff
@@ -41,9 +51,21 @@ export function registerEndpoints(
         })
       }
     },
-    async (bazService, req, res) => {
+    async (req, res) => {
+      log.info('handle Baz route');
       const { params, query } = req;
 
+      log.info('create Baz Service instance');
+      // TODO: but BazService needs request's headers
+      // the validated object may not have this now
+      // This is where BazService's underlying elasticsearch service
```

Comment:
> TODO: but BazService needs request's headers 
> the validated object may not have this now

Sorry, I don't really get this part, what is `validated object` and why it may not have `headers` after validation? Could you please elaborate a bit on that?

Reply:
@azasypkin the `req` object that's available in the route handler function contains only the things that have been validated by the `validate:` key in the lines above:
```js
      validate: {
        params: object({
          type: string()
        }),
        query: object({
          page: maybe(
            number({
              min: 1
            })
          ),
          per_page: maybe(
            number({
              min: 1
            })
          )
        })
      }
```
Right now that doesn't include request headers. 

Maybe we can make an exception that we pass along request headers just for the time being.

Reply:
I made a note of the above ^^. I'll put up another PR exploring validating headers.